tests/jointcalTestBase.py

```diff
@@ -0,0 +1,311 @@
+
+    @param data (dict) dict of starID:list(measurement deltas for that star)
+
+    @return (np.array) the rms for each starID (no guaranteed order)
+    """
+    result = np.empty(len(data))
+    for i, k in enumerate(data):
+        result[i] = np.sqrt(np.mean(np.array(data[k])**2))
+    return result
+
+
+def rms_total(data):
+    """
+    Compute the total rms across all stars.
+
+    @param data (dict) dict of starID:list(measurement deltas for that star)
+
+    @return (int) the total rms across all stars
+    """
+    n = 0
+    total = 0
+    for i, k in enumerate(data):
+        total += sum(np.array(data[k])**2)
+        n += len(data[k])
+    return np.sqrt(total/n)
+
+
+def plot_all_wcs_deltas(plt, dataRefs, visitCatalogs, old_wcss):
+    """Various plots of the difference between old and new Wcs."""
+
+    plot_all_wcs_quivers(plt, dataRefs, visitCatalogs, old_wcss)
+
+    # NOTE: uncomment this block to get per-ccd plots of the "distortion map".
+    # for i, ref in enumerate(dataRefs):
+    #     md = ref.get('calexp_md')
+    #     plot_wcs(plt, old_wcss[i], ref.get('wcs').getWcs(),
+    #              dim=(md.get('NAXIS1'), md.get('NAXIS1')),
+    #              center=(md.get('CRVAL1'), md.get('CRVAL2')), name='dataRef %d'%i)
+
+
+def wcs_convert(xv, yv, wcs):
+    """Convert two arrays of x/y points into an on-sky grid."""
+    xout = np.zeros((xv.shape[0], yv.shape[0]))
+    yout = np.zeros((xv.shape[0], yv.shape[0]))
+    for i, x in enumerate(xv):
+        for j, y in enumerate(yv):
+            sky = wcs.pixelToSky(x, y).toFk5()
+            xout[i, j] = sky.getRa()
+            yout[i, j] = sky.getDec()
+    return xout, yout
+
+
+def plot_all_wcs_quivers(plt, dataRefs, visitCatalogs, old_wcss):
+    """Make quiver plots of the WCS deltas for each CCD in each visit."""
+
+    for cat in visitCatalogs:
+        fig = plt.figure()
+        fig.set_tight_layout(True)
+        ax = fig.add_subplot(111)
+        for old_wcs, ref in zip(old_wcss, dataRefs):
+            if ref.dataId['visit'] == cat:
+                md = ref.get('calexp_md')
+                Q = plot_wcs_quivers(ax, old_wcs, ref.get('wcs').getWcs(),
+                                     dim=(md.get('NAXIS1'), md.get('NAXIS2')))
+                # TODO: add CCD bounding boxes to plot once DM-5503 is finished.
+                # TODO: add a circle for the full focal plane.
+        ax.quiverkey(Q, 0.9, 0.95, 0.1/arcsec_per_radian, '0.1 arcsec', coordinates='figure')
+        plt.xlabel('RA')
+        plt.ylabel('Dec')
+        plt.title('visit: {}'.format(cat))
+
+
+def plot_wcs_quivers(ax, wcs1, wcs2, dim=(1000, 1000)):
+    """Plot the delta between wcs1 and wcs2 as vector arrows."""
+
+    x = np.linspace(0, dim[0], 50)
+    y = np.linspace(0, dim[0], 50)
+    x1, y1 = wcs_convert(x, y, wcs1)
+    x2, y2 = wcs_convert(x, y, wcs2)
+    u = x2-x1
+    v = y2-y1
+    Q = ax.quiver(x1, y1, u, v, units='x', pivot='tail', scale=1e-3, width=5e-6)
+    return Q
```

Comment:
`return ax.quiver(......)` because you're never gonna want to look at it.
I don't think you should hard-code default `dims` because there's no good default.